src/mumble/pluginAPI/Plugin.h

```diff
@@ -0,0 +1,203 @@
+	/// be copied for further usage though.
+	///
+	/// @returns A pointer to the description (encoded as a C-String)
+	PLUGIN_EXPORT const char* getDescription();
+
+	/// Provides the MumbleAPI struct to the plugin. This struct contains function pointers that can be used
+	/// to interact with the Mumble client. It is up to the plugin to store this struct somewhere if it wants to make use
+	/// of it at some point.
+	///
+	/// @param api The MumbleAPI struct
+	PLUGIN_EXPORT void registerAPIFunctions(const struct MumbleAPI *api);
+
+	/// Registers the ID of this plugin. This is the ID Mumble will reference this plugin with and by which this plugin
+	/// can identify itself when communicating with Mumble.
+	///
+	/// @param id The ID for this plugin
+	PLUGIN_EXPORT void registerPluginID(uint32_t id);
+	
+	/// Gets the feature set of this plugin. The feature set is described by bitwise or'ing the elements of the PluginFeature enum
+	/// together.
+	///
+	/// @returns The feature set of this plugin
+	PLUGIN_EXPORT uint16_t getPluginFeatures();
+
+	/// Requests this plugin to deactivate the given (sub)set of provided features.
+	/// If this is not possible, the features that can't be deactivated shall be returned by this function.
+	/// 
+	/// Example (check if FEATURE_POSITIONAL shall be deactivated):
+	/// @code
+	/// if ((features & FEATURE_POSITIONAL) == FEATURE_POSITIONAL) {
+	/// 	// positional shall be deactivated
+	/// };
+	/// @endcode
+	///
+	/// @param features The feature set that shall be deactivated
+	/// @returns The feature set that can't be disabled (bitwise or'ed). If all requested features can be disabled, return
+	/// 	FEATURE_NONE. If none of the requested features can be disabled return the unmodified features parameter.
+	PLUGIN_EXPORT uint32_t deactivateFeatures(uint32_t features);
+
+	
+	// -------- Positional Audio --------
+
+	/// Retrieves the positional audio data. If no data can be fetched, set all float-vectors to 0 and return false.
+	///
+	/// @param[out] avatar_pos A float-array of size 3 representing the cartesian position of the player/avatar in the ingame world.
+	/// 	One unit represents one meter of distance.
+	/// @param[out] avatar_front A float-array of size 3 representing the cartesian direction-vector of the player/avatar ingame (where it
+	/// 	is facing). One unit represents one meter of distance.
+	/// @param[out] avatar_top A float-array of size 3 representing <TODO>.One unit represents one meter of distance.
+	/// @param[out] camera_pos A float-array of size 3 representing the cartesian position of the camera in the ingame world.
+	/// 	One unit represents one meter of distance.
+	/// @param[out] camera_front A float-array of size 3 representing the cartesian direction-vector of the camera ingame (where it
+	/// 	is facing). One unit represents one meter of distance.
+	/// @param[out] camera_top A float-array of size 3 representing <TODO>.One unit represents one meter of distance.
```

Comment:
@davidebeatrici  Could you give me a description of what the "top-arrays" should represent?

Reply:
https://github.com/mumble-voip/mumble/blob/5f744ed9710452364f7681437be8bf2ae190c41b/plugins/mumble_plugin.h#L147-L152

Reply:
Yeah that's the part I don't quite understand xD
If `avatar_front` is where the character is looking, is `avatar_top` the vector from its toes to its head? Why would that be useful for positional audio :thinking:

Reply:
We're actually using neither in our current mixer code, we only take the position and the three camera vectors into account.

Reply:
Okay ^^

> s avatar_top the vector from its toes to its head?

Is that correct though? I just want the description to be as correct as possible :)

Reply:
Yep, it's correct!